.travis.yml

```diff
@@ -46,6 +46,10 @@ jobs:
         # Install and run `rubocop`
         - gem install rubocop
         - rubocop -d
+        # Run `shellcheck` (already pre-installed in Travis)
+        - shellcheck --version
+        - git ls-files | grep '\.sh$\|\.bash$\|\.ksh$'
```

Comment:
Do you know that `git ls-files` accepts patterns itself as [pathspecs](https://kgrz.io/git-intro-to-pathspec.html), so
```
git ls-files -- '*.sh' '*.bash' '*.ksh' | xargs shellcheck
```
would work equally well here (and reads better imho)?

Reply:
@dafyddj Good to know, looks like it works without the quotes as well.  Will have to update the `salt-lint` invocation as well when applying this.

Reply:
Applied, thanks.

Reply:
> @dafyddj Good to know, looks like it works without the quotes as well. Will have to update the `salt-lint` invocation as well when applying this.

To my knowledge it shouldn't work _correctly_ without quotes. How did you test that?

Reply:
@dafyddj The simplest check, comparing the output of both without the pipe to `shellcheck` (or `salt-lint` in the example below), i.e.:

```bash
fail2ban-formula$  git ls-files -- '*.sls' '*.jinja' '*.j2' '*.tmpl' '*.tst'
fail2ban/config.sls
fail2ban/deprecated.sls
fail2ban/init.sls
fail2ban/map.jinja
fail2ban/ng/config.sls
fail2ban/ng/files/config.jinja
fail2ban/ng/init.sls
fail2ban/ng/install.sls
fail2ban/ng/map.jinja
fail2ban/ng/service.sls
test/salt/states/misc/fake_log_files.sls
fail2ban-formula$  git ls-files -- *.sls *.jinja *.j2 *.tmpl *.tst
fail2ban/config.sls
fail2ban/deprecated.sls
fail2ban/init.sls
fail2ban/map.jinja
fail2ban/ng/config.sls
fail2ban/ng/files/config.jinja
fail2ban/ng/init.sls
fail2ban/ng/install.sls
fail2ban/ng/map.jinja
fail2ban/ng/service.sls
test/salt/states/misc/fake_log_files.sls
```

Also tested this out with the pipes to both `shellcheck` and `salt-lint` respectively.

Reply:
@dafyddj Never mind, I've already started propagating the quotes using:

* https://github.com/myii/ssf-formula/pull/107

It should be done within the next few minutes.  Thanks for getting back to me.

Reply:
I don't deny that your test works but there's something strange going on here for me, and I'm trying to work out what that is.

Reply:
Ok, your test only worked by accident. :)

From [gnu](https://www.gnu.org/software/bash/manual/html_node/Filename-Expansion.html)

> If no matching filenames are found, and the shell option nullglob is disabled, the word is left unchanged.

So because no files happen to match in the root directory of `fail2ban-formula` `git ls-files` receives the patterns themselves, and works as expected. The results would not be complete if a matching file existed in the root directory.

To be honest, I've never come across this situation before so it was handy to learn more about Bash!

Reply:
Funny, because it worked in the first few formulas I tried!  At least you got to the bottom of that, thanks for sharing.

Actually, this overlaps with the example here, using `shopt` to keep `nullglob` enabled:

* https://github.com/koalaman/shellcheck/wiki/TravisCI